django/db/models/indexes.py

```diff
@@ -51,12 +58,26 @@ def check_name(self):
             self.name = 'D%s' % self.name[1:]
         return errors
 
+    def _get_condition_sql(self, model, schema_editor):
+        if self.condition is not None:
+            query = Query(model=model)
+            query.add_q(self.condition)
+            compiler = query.get_compiler(connection=schema_editor.connection)
+            # Only the WhereNode is of interest for the partial index
+            sql, params = query.where.as_sql(compiler=compiler, connection=schema_editor.connection)
+            # The base schema editor does the same map on the params, but since it's
+            # handled outside of that class, the work is done here
+            return ' WHERE ' + (sql % tuple(map(schema_editor.quote_value, params)))
+        return ''
```

Comment:
I think this would be slightly cleaner:

```python
        if self.condition is None:
            return ''
        query = Query(model=model)
        query.add_q(self.condition)
        compiler = query.get_compiler(connection=schema_editor.connection)
        # Only the WhereNode is of interest for the partial index
        sql, params = query.where.as_sql(compiler=compiler, connection=schema_editor.connection)
        # The base schema editor does the same map on the params, but since it's
        # handled outside of that class, the work is done here
        return ' WHERE ' + (sql % tuple(map(schema_editor.quote_value, params)))
```

Reply:
:+1: